assertions/manager.go

```diff
@@ -98,61 +112,105 @@ func WithDangerousReadyToPost() Opt {
+//
+// This is the interval the assertion manager will wait between attempts to
+// persist information about which assertions can be confirmed to the parent
+// chain.
+func WithConfirmationInterval(t time.Duration) Opt {
+	return func(m *Manager) {
+		m.times.confInterval = t
+	}
+}
+
+// WithAverageBlockCreationTime overrids the default average block creation
+// time.
+//
+// The average block cretion time is used by the assertion manager to emit
+// warnings if the parent chain hasn't had any new blocks for considerably
+// longer than this expected delay.
+func WithAverageBlockCreationTime(t time.Duration) Opt {
+	return func(m *Manager) {
+		m.times.avgBlockTime = t
+	}
+}
+
 // NewManager creates a manager from the required dependencies.
 func NewManager(
 	chain protocol.AssertionChain,
-	stateProvider l2stateprovider.Provider,
-	backend bind.ContractBackend,
-	challengeManager types.ChallengeManager,
-	rollupAddr common.Address,
-	challengeManagerAddr common.Address,
+	execProvider l2stateprovider.ExecutionProvider,
 	validatorName string,
-	pollInterval,
-	assertionConfirmationAttemptInterval time.Duration,
-	stateManager l2stateprovider.ExecutionProvider,
-	postInterval time.Duration,
-	averageTimeForBlockCreation time.Duration,
-	apiDB db.Database,
+	mode types.Mode,
 	opts ...Opt,
 ) (*Manager, error) {
-	if pollInterval == 0 {
-		return nil, errors.New("assertion scanning interval must be greater than 0")
-	}
-	if assertionConfirmationAttemptInterval == 0 {
-		return nil, errors.New("assertion confirmation attempt interval must be greater than 0")
-	}
 	m := &Manager{
-		chain:                       chain,
-		apiDB:                       apiDB,
-		backend:                     backend,
-		stateProvider:               stateProvider,
-		challengeCreator:            challengeManager,
-		challengeReader:             challengeManager,
-		rollupAddr:                  rollupAddr,
-		challengeManagerAddr:        challengeManagerAddr,
-		validatorName:               validatorName,
-		pollInterval:                pollInterval,
-		confirmationAttemptInterval: assertionConfirmationAttemptInterval,
-		forksDetectedCount:          0,
-		challengesSubmittedCount:    0,
-		assertionsProcessedCount:    0,
-		postInterval:                postInterval,
-		submittedAssertions:         threadsafe.NewLruSet[common.Hash](1000, threadsafe.LruSetWithMetric[common.Hash]("submittedAssertions")),
-		averageTimeForBlockCreation: averageTimeForBlockCreation,
+		chain:                    chain,
+		apiDB:                    nil,
+		backend:                  chain.Backend(),
+		execProvider:             execProvider,
+		rollupAddr:               chain.RollupAddress(),
+		validatorName:            validatorName,
+		times:                    defaultTimings,
+		forksDetectedCount:       0,
+		assertionsProcessedCount: 0,
+		submittedAssertions:      threadsafe.NewLruSet(1000, threadsafe.LruSetWithMetric[common.Hash]("submittedAssertions")),
```

Comment:
For now, I'm bumping this to 1500 which is 15.6 days at 15 minutes per assertion.
But ... if we really want to avoid cache misses for 2 weeks, we should probably come up with this number based on the input parameters like average block creation time, and maybe the length of a challenge window?

Reply:
I've opened https://linear.app/offchain-labs/issue/NIT-2955/calculate-the-lru-cache-sizes-instead-of-hardcoding-them-to-1500 to track the change to dynamically calculated LRU cache sizes. I'm resolving this comment for now, and we can address this issue in that ticket.